File: examples/analyze-passage.mjs
// Node.js usage example.
//
// Requires the native addon to be built first: `npm run build:debug`.
//
//   node examples/analyze-passage.mjs

import { computeReadability, checkGrammar, listStructureTemplates } from '../index.mjs';

const passage =
  'Maren stood at the edge of the dock, watching the last ferry pull away ' +
  'without her. The water slapped against the pilings, cold and indifferent.';

const readability = computeReadability(passage);
console.log(
  `readability: fkgl=${readability.fkgl.toFixed(1)} (${readability.wordCount} words, ${readability.sentenceCount} sentences)`
);

const findings = checkGrammar(passage);
console.log(`grammar: ${findings.length} finding(s)`);

console.log(`available structure templates: ${listStructureTemplates().join(', ')}`);

File: examples/analyze_passage.rs
//! Minimal end-to-end example: run a few analyzers over one passage.
//!
//! ```sh
//! cargo run --example analyze_passage
//! ```

use narrative_lens::craft::{grammar, readability};
use narrative_lens::structure::structure;

fn main() {
    let passage = "Maren stood at the edge of the dock, watching the last ferry \
        pull away without her. The water slapped against the pilings, cold and \
        indifferent. She had missed it on purpose, though she was not yet ready \
        to admit that to herself.";

    let readability = readability::compute_readability(passage);
    println!(
        "readability: fkgl={:.1} ({} words, {} sentences)",
        readability.fkgl, readability.word_count, readability.sentence_count
    );

    let findings = grammar::check(passage);
    println!("grammar: {} finding(s)", findings.len());

    let templates = structure::list_templates();
    println!("available structure templates: {}", templates.join(", "));
}
